plugins/techdocs-backend/src/service/router.ts

```diff
@@ -17,18 +17,76 @@ import { Logger } from 'winston';
 import Router from 'express-promise-router';
 import express from 'express';
 import Knex from 'knex';
+import fetch from 'node-fetch';
+import { Config } from '@backstage/config';
+import path from 'path';
+import Docker from 'dockerode';
+import {
+  GeneratorBuilder,
+  PreparerBuilder,
+  PublisherBase,
+  LocalPublish,
+} from '../techdocs';
+import { Entity } from '@backstage/catalog-model';
 
 type RouterOptions = {
+  preparers: PreparerBuilder;
+  generators: GeneratorBuilder;
+  publisher: PublisherBase;
   logger: Logger;
   database?: Knex; // TODO: Make database required when we're implementing database stuff.
+  config: Config;
+  dockerClient: Docker;
```

Comment:
This abstraction isn't that useful since it's more or less the concrete `dockerode` type and only really leaves room of mocking and decoration. I see it's copied from the scaffolder anyway though, but it could be worth spending some effort here to see if it's possible do depend on something more narrow. Something like a DockerRunner interface.

Reply:
Can't find anything better to use in the type declaration. Will revisit some of these things soon and i can have a deeper dive into a better solution then.

Reply:
Yeah would be an interface defined by the plugin that's tailored to the needs here. Doesn't have to directly match the dockerode client either, you could wrap that a bit if needed.